common/persistence/health_signal_aggregator.go

```diff
@@ -153,18 +162,17 @@ func (s *HealthSignalAggregatorImpl) emitMetricsLoop() {
 	}
 }
 
-// TODO: uncomment when adding dynamic rate limiter
-//func isUnhealthyError(err error) bool {
-//	if err == nil {
-//		return false
-//	}
-//	switch err.(type) {
-//	case *ShardOwnershipLostError,
-//		*AppendHistoryTimeoutError,
-//		*TimeoutError:
-//		return true
-//
-//	default:
-//		return false
-//	}
-//}
+func isUnhealthyError(err error) bool {
+	if err == nil {
+		return false
+	}
+	switch err.(type) {
+	case *AppendHistoryTimeoutError,
+		*TimeoutError,
+		*serviceerror.DeadlineExceeded:
```

Comment:
oh I actually mean context.DeadlineExceeded, I am not sure if we always convert context.DeadlineExceeded to persistence.TimeoutError. Looks like the case for cassandra but I don't the logic of sql. 
maybe use `common.IsContextDeadlineExceededErr` to cover both context.DeadlineExceeded and serviceerrror.DeadlineExceeded. also keep the type check for persistence.TimeoutError.